### PR TITLE
Test revocation notification webhook certificate configuration

### PR DESCRIPTION
**Pull request by ansasaki**
The added test verifies the behavior of the revocation notification webhook when configured with valid certificate, invalid certificate, no certificate, and system-wide installed CA certificate.

### CODE REVIEW

File: functional/webhook-certificate-on-localhost/payload-module/autorun.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
```
Comment (kkaarreell) [inline functional/webhook-certificate-on-localhost/payload-module/autorun.sh:1]: Please remove all this payload-module related stuff, it is not needed.

File: functional/webhook-certificate-on-localhost/test.sh [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                rlRun "sleep 500 | openssl s_server -debug -nocert -port ${WEBHOOK_SERVER_PORT} &> ${WEBHOOK_SERVER_LOG} &"
+                WEBHOOK_SERVER_PID=$!
+            else
+                rlRun "sleep 500 | openssl s_server -debug -cert $(x509Cert "${WEBHOOK}") -key $(x509Key "${WEBHOOK}") -port ${WEBHOOK_SERVER_PORT} &> ${WEBHOOK_SERVER_LOG} &"
+                WEBHOOK_SERVER_PID=$!
+            fi
+        rlPhaseEnd
+
+        rlPhaseStartTest "Add keylime agent (WEBHOOK=${WEBHOOK})"
+            REVOCATION_SCRIPT_TYPE=$( limeGetRevocationScriptType )
```
Comment (kkaarreell) [inline functional/webhook-certificate-on-localhost/test.sh:153]: use hardcoded payload-script.

File: functional/webhook-certificate-on-localhost/main.fmf [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+component:
+  - keylime
+test: ./test.sh
+framework: beakerlib
+tag:
+  - CI-Tier-1
+require:
+  - expect
+  - library(openssl/certgen)
+  - nmap-ncat
```
Comment (kkaarreell) [inline functional/webhook-certificate-on-localhost/main.fmf:23]: nmap is not needed, right?

File: functional/webhook-certificate-on-localhost/test.sh [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+            limeExtendNextExcludelist "$TESTDIR"
+            limeCreateTestPolicy
+        rlPhaseEnd
+    done
+
+    rlPhaseStartCleanup "Do the keylime cleanup"
+        rlRun "rm -f /var/tmp/test_payload_file"
+        rlRun "limeStopAgent"
+        rlRun "limeStopRegistrar"
+        rlRun "limeStopVerifier"
```
Comment (kkaarreell) [inline functional/webhook-certificate-on-localhost/test.sh:215]: should be already stopped

File: functional/webhook-certificate-on-localhost/test.sh [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+        read -r WEBHOOK EXPECTED_RESULT WEBHOOK_SERVER_PORT EXPECTED_ERROR <<< "${i}"
+
+        # This case is to test that the verifier is including the system-wide
+        # installed certificates when verifying the revocation notification
+        # webhook certificate
+        if [ "${WEBHOOK}" = "installed" ]; then
+            rlPhaseStartTest "Install CA certificate on system-wide store, and run on port ${WEBHOOK_SERVER_PORT}"
+                rlRun "limeUpdateConf revocations webhook_url https://localhost:${WEBHOOK_SERVER_PORT}"
+                # Install the "bad" CA certificate on system-wide trust store
+                rlRun "cp $(x509Cert bad-ca) /etc/pki/ca-trust/source/anchors/keylime-ca.crt"
```
Comment (kkaarreell) [inline functional/webhook-certificate-on-localhost/test.sh:129]: could you rename `keylime-ca.crt` to `webhook-ca.crt`? I find keylime a bit confusing here.

File: functional/webhook-certificate-on-localhost/main.fmf (hunk 3 above)
Comment (ansasaki) [reply functional/webhook-certificate-on-localhost/main.fmf:23]: Correct, I'll drop it

**Review by kkaarreell — APPROVED** [review]

**Comment by kkaarreell** [comment]
Test failures are unrelated. 
For C10S it is failing due to https://issues.redhat.com/browse/RHEL-55991
On F41 it is failing as we do not have keylime-agent-rust copr build.
Rawhide failure is a known bug.

**Pull request merged**